### PR TITLE
First pass balena device

### PR DESCRIPTION
**Pull request by hardillb**
## Description

Describe using Balena to deploy the device agent

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

**Comment by hardillb** [comment]
@Yndira-FlowForge I've just grabbed the Balena logo for the header image, but it would look better if our Logo as well, would you be able to come up with something?

**Comment by Yndira-FlowForge** [comment]
Hi @hardillb, as this image is also displayed as the blog tile, it needs to have a specific size. Could you please create an Art Request issue?

File: src/blog/2023/11/device-agent-balens.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+To do this you install the Balena OS on the devices, this is a stripped down Linux distribution that includes a client that connects back to Balena's platform and creates a VPN tunnel. As well as the Balena client it includes Docker and users can select containers to push to the devices.
+
+
+These Docker container are hosted on Balena's own container registry and are built by doing a git push to Balena's git server.
+
+##Building FlowFuse Device Agent for Balena
+
+We already build a FlowFuse Device Agent Docker container so it was pretty simple to modify the existing `Dockerfile` for Balena.
+
+```
```
Comment (ZJvandeWeg) [inline src/blog/2023/11/device-agent-balens.md:27]:
```suggestion
```Dockerfile
```

File: src/blog/2023/11/device-agent-balens.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+To do this you install the Balena OS on the devices, this is a stripped down Linux distribution that includes a client that connects back to Balena's platform and creates a VPN tunnel. As well as the Balena client it includes Docker and users can select containers to push to the devices.
+
+
+These Docker container are hosted on Balena's own container registry and are built by doing a git push to Balena's git server.
+
+##Building FlowFuse Device Agent for Balena
+
+We already build a FlowFuse Device Agent Docker container so it was pretty simple to modify the existing `Dockerfile` for Balena.
+
+```Dockerfile
```
Comment (hardillb) [inline src/blog/2023/11/device-agent-balens.md:27]:
````suggestion
```
````

File: src/blog/2023/11/device-agent-balena.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+These Docker container are hosted on Balena's own container registry and are built by doing a git push to Balena's git server.
+
+We I have a GitHub repository with all the required files [here](https://github.com/FlowFuse/balena-device-agent), this include a one click deploy button to allow you to quickly try this out.
+
+## Building FlowFuse Device Agent for Balena
+
+We already build a FlowFuse Device Agent Docker container so it was pretty simple to modify the existing `Dockerfile` for Balena.
+
+```
```
Comment (ZJvandeWeg) [inline src/blog/2023/11/device-agent-balena.md:28]: Tried this locally this time:

```suggestion
```docker
```

File: src/blog/2023/11/device-agent-balena.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+ 2. Adding a custom `entrypoint.sh`. This is to ensure that the hostname seen in the container matches the Balena device name, making it easier to match it up with what is seen in the FlowFuse application. It also generates the configuration file from the passed in environment variable (see [below](#configuring-devices))
+
+As well as the `Dockerfile` there is also a `docker-compose.yml` because Balena applications can be made up of multiple services packaged as container. In this case we just need a single container but the compose file contains all the information about what ports to expose and what volumes need creating to persist state.
+
+## Configuring Devices
+
+The FlowFuse Device agent can be configured in 3 ways and the Balena deployment supports all 3.
+
+ 1. If no configuration file is provided then the Device Agent will start up and launch a simple web application that allows you to upload a configuration file. This is the default behaviour for the Balena deployment.
```
Comment (ZJvandeWeg) [inline src/blog/2023/11/device-agent-balena.md:53]: Isn't there a timer that kills the UI after 30 minutes?

File: src/blog/2023/11/device-agent-balena.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+ 2. You can provide a configuration file that is provided by the FlowFuse application when you create a new Device. This file contains the unique identifiers for the Device and details of where to find the FlowFuse Application. This file can be provided to a Balena device by adding a device specific environment variable as described below.
+
+ 3. You can provide a fleet of devices with a configuration file that contains details of where to find the FlowFuse application and a Provisioning token. Multiple Devices can all have the same Provisioning token and this will cause them to connect to the FlowFuse application on first start up and create a new Device bound to an existing team (and optionally an Application or Instance). This file can be passed to Balena devices by way of a Fleet wide environment variable. 
+ You can create a Provisioning Token file under the Team -> Settings page on the Devices tab.
+
+Because the file mentioned in options 2 & 3 is multi line it needs to be base64 encoded, you can do this with the following
+
+$ base64 -w 0 device.yml
```
Comment (ZJvandeWeg) [inline src/blog/2023/11/device-agent-balena.md:62]:
```suggestion
`$ base64 -w 0 device.yml`
```

**Review by ZJvandeWeg — APPROVED** [review]

**Comment by ZJvandeWeg** [comment]
@hardillb Merge when ever you're ready, and please schedule posts (or ask for someone to do this) on social media.

File: src/blog/2023/11/device-agent-balena.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
```
Comment (hardillb) [inline src/blog/2023/11/device-agent-balena.md:62]:
````suggestion
```bash
$ base64 -w 0 device.yml
```
````

File: src/blog/2023/11/device-agent-balena.md (hunk 4 above)
Comment (hardillb) [reply src/blog/2023/11/device-agent-balena.md:53]: The default is 10mins. I will add a comment

**Pull request merged**